traversal/BUILD

```diff
@@ -42,6 +42,7 @@ native_java_libraries(
         "@vaticle_typedb_common//:common",
         "@vaticle_typeql//java/common:common",
         "@vaticle_typeql//java/pattern:pattern",
+        "@vaticle_typeql//java/query:query",
```

Comment:
beause we can construct `Modifier.Filter` and `Modifier.Sorting` from the TypeQL representations, we have to allow accessing the TypeQL construct from the `//traversal` package. This seems acceptable, especially since we do something similar for `Identifier` already